### ISSUE TITLE
v12.5 does no longer work offline for me

### ISSUE THREAD

**Issue by RealVegOs** [bailuk/AAT#160]
I have a maps folder on sd card. If I select this for offline maps, the app cannot read the files inside. I think this is a matter of map theme Elevate. If I select DEFAULT or OSMARENDER, it is said, they do not exist. I uninstalled AAT completely, downloaded new map and Elevate files, but after installing AAT again, it still can not read the offline maps. Please try yourself, by installing AAT form F-Droid, not from your repo. And please excuse that I cannot explain this stuff better. Thanks much!

**Comment by RealVegOs** [comment]
They have a black background. I'll look after the cache setting. THX & Happy Father's Day!

**Comment by RealVegOs** [comment]
Caching und Offline in map setting is activated.

**Comment by bailuk** [comment]
This issue is hard to reproduce for me. After I upgraded from f-droid,  offline maps did not work. Then after changing some of the setting it worked again.

For testing there is a `aat-android-debug.apk` distribution that can be installed in parallel to the official release. 

It is available from the [release page](https://github.com/bailuk/AAT/releases/tag/v1.25) and from [automated builds](https://github.com/bailuk/AAT/actions) and is inside `aat-android.zip`

Some things to check:
- Is there a blue `"is readonly" note for maps and and themes in the preferences?
- Is permission for "Storage" enabled (android app info)?

**Comment by bailuk** [comment]
> Caching und Offline in map setting is activated.

Of course map tiles need to be regenerated and there must be some images in the map cache (after using the map).  To regenerate a preview image: `click on image` -> `Reload preview`. If you want to generate all previews you need to delete the `cache` directory inside the `activity` directory. Like `aat_data\activity0\cache`

**Comment by RealVegOs** [comment]
THX! - Later, I'm on my way.

**Comment by RealVegOs** [comment]
> This issue is hard to reproduce for me. After I upgraded from f-droid, offline maps did not work. Then after changing some of the setting it worked again.
> 
> For testing there is a `aat-android-debug.apk` distribution that can be installed in parallel to the official release.
> 
> It is available from the [release page](https://github.com/bailuk/AAT/releases/tag/v1.25) and from [automated builds](https://github.com/bailuk/AAT/actions) and is inside `aat-android.zip`
> 
> Some things to check:
> 
>     * Is there a blue `"is readonly" note for maps and and themes in the preferences?
> 
>     * Is permission for "Storage" enabled (android app info)?

I installed the debug version. (I already had it, but did not find it...) I have all my stuff on a sd card. So I have to select or copy the paths in general an map settings.
![Screenshot_20240510-190356](https://github.com/bailuk/AAT/assets/7726676/7456ffdd-b2e7-4769-84f8-eca138bb2e78)
![Screenshot_20240510-190546](https://github.com/bailuk/AAT/assets/7726676/ea1bb1bd-437f-4de1-a93d-e3a162e34695)

The map path is correct  but as you see on the second screenshot it is read only. If I open the map view, I get an error that the map is not readable. I think that's a matter of Elevate. It works in v12.4 with Elevate52. BTW, most preview maps are back now in the old version.

Alas, I can't send you crash reports, because I have no email client on my Tablet. It's a WiFi Tablet only. Crashes occured by playing with settings, changing paths etc. 

Cheers https://www.youtube.com/watch?v=Rf1T6S3qzW8

**Comment by mingodad** [comment]
Even the 1.26 doesn't work for me on android 14 Redmi Xiaomi HyperOS 1.0.10.0, **I've installed the 1.24 debug apk and I can see my offline maps**.

**Comment by RealVegOs** [comment]
Same for me! I also went back from 1.26 back to 1.24. Latest debug version does not work too. Again, I think 
that's an issue with Elevate. And of course Android is the biggest shit on earth! You even can't get the real path of files unless you root your device, which ends the warranty in most cases.

**Comment by RealVegOs** [comment]
> Some things to check:
>    Is there a blue `"is readonly" note for maps and and themes in the preferences?
>    Is permission for "Storage" enabled (android app info)?

Stupid question, where do I find this permission setting for Android 13?  I found  App Info. There is "Storage and Cache" but there is nothing to enable.

![grafik](https://github.com/user-attachments/assets/69a62c76-d14b-454c-86c5-50833662943f)

**Comment by RealVegOs** [comment]
I played a bit more but with no success. If I create a sub folder suddenly read and write access for it is shown in option. But opening card view still yields an error, no files found. If I give the full path to the map, cannot read file error occurs. I also was not able to grant read and write or manage external storage permission with adb. Nothing changed even without any adb error. So sadly I still have to stick to v1.2.4 Please, please, please!

**Comment by bailuk** [comment]
Maybe your using an old Map format that isn't supported anymore by MapsForge. 
You should use V5 Maps as described here: https://github.com/bailuk/AAT/blob/master/doc/manual.md

**Comment by RealVegOs** [comment]
THX. Alas no chance. I downloaded the latest map for Berlin and the latest Elements but I still get no maps found.

**Comment by RealVegOs** [comment]
Perhaps this may help:
I have the maps directory on my SD card, which works well for v1.24
For debug v1.26 I have a maps_debug directory on my SD card. But as the screenshots above show, putting the right stuff for v1.26 to maps did not work.

**Comment by bailuk** [comment]
- Application will open a dialog to ask user for full access to sdcard (on first application start or on selecting `request permissions` in settings)
- Added section in `Preferences->General/System` to check and request permissions (does only check for some permissions for now)
- Also add permission for notification on newer Android versions (needs additional configuration in android App Permission settings)
- Tested in emulator: Pixel 6, API 33, Android 13.
- This will be in `v.1.27` - reopen issue or add new issue if something should not work

**Comment by RealVegOs** [comment]
I keep thousand fingers crossed that this works.
